python/pyspark/_typing.pyi

```diff
@@ -21,11 +21,14 @@ from typing_extensions import Protocol
 
 F = TypeVar("F", bound=Callable)
 T = TypeVar("T", covariant=True)
+U = TypeVar("U", bound="SupportsIAdd")
```

Comment:
Why would add bound for protocol?

Reply:
Should we not add bound `SupportsIAdd` for `U` which can make a constraint for types `U` can be ?